src/commands/publish/mod.rs

```diff
@@ -89,7 +89,26 @@ fn publish_script(user: &GlobalUser, target: &Target) -> Result<(), failure::Err
 fn upload_buckets(target: &Target, user: &GlobalUser) -> Result<(), failure::Error> {
     for namespace in &target.kv_namespaces() {
         if let Some(bucket) = &namespace.bucket {
+            if bucket.is_empty() {
+                failure::bail!(
+                    "{} You need to specify a bucket directory in your wrangler.toml",
+                    emoji::WARN
+                )
+            }
             let path = Path::new(&bucket);
+            if !path.exists() {
+                failure::bail!(
+                    "{} bucket directory \"{}\" does not exist",
+                    emoji::WARN,
+                    path.to_string_lossy()
+                )
+            } else if !path.is_dir() {
```

Comment:
some places we use metadata, some places we use path. curious if metadata would be better since i suspect it comes from the OS rather than being inferred by the string contents.

Reply:
we should do a review and audit of this- filed an issue: https://github.com/cloudflare/wrangler/issues/696

Reply:
made a todo for later.